src/components/Common/EmojiPickerIconButton.tsx

```diff
@@ -0,0 +1,74 @@
+import { Box, IconButton } from "@mui/material";
+import EmojiEmotionsOutlinedIcon from "@mui/icons-material/EmojiEmotionsOutlined";
+import { useState, useEffect, useRef } from "react";
+import EmojiPicker, { EmojiClickData, EmojiStyle } from "emoji-picker-react";
+
+type EmojiPickerIconButtonProps = {
+  onEmojiClick: (emoji: EmojiClickData) => void;
+  pickerHeight?: number;
+  pickerPosition?: "top" | "bottom";
+  pickerWidth?: number;
+};
+
+const styles = {
+  emojiContainer: { position: "absolute", zIndex: 1 },
+  emojiPicker: {
+    "--epr-emoji-size": "25px",
+  },
+};
+
+const EmojiPickerIconButton = ({
+  onEmojiClick,
+  pickerHeight = 450,
+  pickerPosition = "bottom",
+  pickerWidth = 350,
+}: EmojiPickerIconButtonProps) => {
+  const emojiContainerRef = useRef<HTMLDivElement>(null);
+  const emojiIconButtonRef = useRef<HTMLButtonElement>(null);
+  const twitterEmojiStyle = EmojiStyle.TWITTER;
+  const [showEmojiPicker, setShowEmojiPicker] = useState(false);
+  const top = pickerPosition === "bottom" ? 150 : -460;
+
+  const handleClick = (event: MouseEvent) => {
+    if (
+      emojiContainerRef.current &&
+      !emojiContainerRef.current.contains(event.target as Node) &&
+      emojiIconButtonRef.current &&
+      !emojiIconButtonRef.current.contains(event.target as Node)
+    ) {
+      setShowEmojiPicker(false);
+    }
+  };
+
+  useEffect(() => {
+    document.body.addEventListener("mousedown", handleClick);
+    return () => {
+      document.removeEventListener("mousedown", handleClick);
+    };
+  }, []);
+
+  return (
+    <>
+      <IconButton
+        ref={emojiIconButtonRef}
+        size="small"
+        onClick={() => setShowEmojiPicker(!showEmojiPicker)}
+      >
+        <EmojiEmotionsOutlinedIcon />
```

Comment:
Yea sure, I can do that. It sucks a little because we defined the color for our icon buttons in the theme to be gray, so we can't use the `color` prop of the icon button. To get around it we have to use the `sx` prop and do some conditional stuff, which is why there's some funky syntax:

```
const styles = {
...
  activeButton: {
    color: "primary.main",
  },
};

...

<IconButton
  ref={emojiIconButtonRef}
  onClick={() => setShowEmojiPicker(!showEmojiPicker)}
  size="small"
  sx={{ ...(showEmojiPicker && styles.activeButton) }}
 >
  <EmojiEmotionsOutlinedIcon />
</IconButton>

```

Reply:
Yeah I agree I think it looks nice. I'm fine with the syntax, it's a bit confusing at first but it makes sense to me and this case is sort of a one off so I think it's whatever